### ISSUE TITLE
Polygons constructor for Nef_polyhedron_2 results in inconsistent state.

### ISSUE THREAD

**Issue by elfprince13** [CGAL/cgal#3449]
## Issue Details

When the `template<class Forward_iterator > Nef_polyhedron_2 (Forward_iterator it, Forward_iterator end, Polygons_tag)` constructor is used on overlapping polygons, the resulting `Nef_polyhedron_2` is in an inconsistent state.

## Source Code
Using the two following polygons as input:
>	3 1 1 2 1 1.5 1.75  
>	9 0 0 3 0 1.85714 2.28571 3 4 2 4 1.5 3 1 4 0 4 1.14286 2.28571  

The code [here](https://github.com/Geopipe/PolygonSimplification/blob/7a1638fc11ec4aa94266e9d2ba4bd960ed8ce09c/include/PolygonSimplification.hpp#L86-L94) results in a `Nef_polyhedron_2` which still contains both polygons in its list of faces.

The code [here](https://github.com/Geopipe/PolygonSimplification/blob/master/include/PolygonSimplification.hpp#L106-L116) exhibits the expected behavior (a single face).

## Environment
> [thomas@Mithlond] PolygonSimplification $ clang --version
> clang version 3.8.1 (tags/RELEASE_381/final)
> Target: x86_64-apple-darwin15.6.0
> Thread model: posix
> InstalledDir: /opt/local/libexec/llvm-3.8/bin
> [thomas@Mithlond] PolygonSimplification $ port installed cgal
> The following ports are currently installed:
>   cgal @4.11_1 (active)
> [thomas@Mithlond] PolygonSimplification $ port installed eigen3
> The following ports are currently installed:
>   eigen3 @3.3.4_0 (active)

**Comment by elfprince13** [comment]
(The simplest fix would be a big flashing warning label: "Don't use this constructor in this way, or your code will break unexpectedly downstream")

**Comment by afabri** [comment]
I am probably too late to help you, but I've put your code in a self contained [nef_polygon.cpp](https://gist.github.com/afabri/c132c24862b6e5f6ab25c844f4f5d298), and when I run it I get the union of the two polygons (the smaller being completely in the larger).   When I run it, it writes "2 faces", but that is normal as we have the unbounded face as well.      So I would tend to close this issue.

**Comment by elfprince13** [comment]
Have there been any changes to Nef_2 since I created this issue? If not I'll dig into your test code more carefully and make sure it captures the necessary behavior.

**Comment by afabri** [comment]
I just verified it with CGAL 4.13.1.
Still let me rephrase your problem to see if I understood you correctly: When you use the constructor that takes a range of simple polygons, the resulting Nef polygon does not represent the union of the input?

**Comment by elfprince13** [comment]
Correct. I have created a [simple example](https://github.com/Geopipe/PolygonSimplification/blob/91ad91f6fb93b725cd2116e20ed35aca7fcdc97a/problem.cpp) to demonstrate this.

The output using CGAL 4.13.1 is:
```
Finished set:
	3 1 1 2 1 1.5 1.75  ( is simple? 1 )
	9 0 0 3 0 1.85714 2.28571 3 4 2 4 1.5 3 1 4 0 4 1.14286 2.28571  ( is simple? 1 )
```

However, a correct output would be:
```
Finished set:
	9 0 0 3 0 1.85714 2.28571 3 4 2 4 1.5 3 1 4 0 4 1.14286 2.28571  ( is simple? 1 )
```

Since the smaller polygon is strictly contained by the larger:
![mwe](https://user-images.githubusercontent.com/2703145/58515409-b980ed80-8172-11e9-9406-2a2e90e5288d.png)

(note that I advance the face iterator once before entering the loop, so the unbounded face is not included)
